src/citation/schema.py

```diff
@@ -13,12 +13,11 @@
 CREATORS_SCHEMA_REGEX = f"({initial_creators_schema_regex})"
 
 
-def generate_json_for_journal(pdf2doi):
-    schema = generate_schema_for_citation(JOURNAL_ARTICLE)
+def generate_json_for_doi(doi):
     json = {}
-    doi_string = pdf2doi["identifier"]
+    schema = generate_schema_for_citation(JOURNAL_ARTICLE)
     open_alex = OpenAlex()
-    result = open_alex.get_data_from_doi(doi_string)
+    result = open_alex.get_data_from_doi(doi)
     for field in schema["required"]:
         mapping_field = OPENALEX_JOURNAL_MAPPING.get(field, "")
```

Comment:
OPENALEX_JOURNAL_MAPPING. Where did we get this from?

Reply:
A lot of this code is originally from https://github.com/ResearchHub/researchhub-backend/pull/1283/files